s3transfer/futures.py

```diff
@@ -170,14 +177,25 @@ def set_exception(self, exception):
                 self._exception = exception
                 self._status = 'failed'
 
-    def result(self):
+    def result(self, sleep_intervals=0.2):
```

Comment:
Not a big fan of having to do sleep intervals. Anyone have good ideas of how we can better handle this and if not what a more appropriate time should be? The reason I have to do this is a straight Event.wait() will consume the KeyboardInterrupt and will not raise it till the event gets set to done, which would be the entirety of the transfer...